### PR TITLE
Status overhaul

### PR DESCRIPTION
**Pull request by milljm**
Refactor of the status bucket system, as well as the QueueManager (RunPBS).

In order to preserve continued PBS functionality, these two features needed to be refactored at the same time.

The status system is a fundamental change, and I fully expect failures from other apps. The common error we will see is:
```
FAILED (deprecated status bucket)
```
This is a simple fix, and will require modifying the offending tester to adjust the positional args of the status, and status message:
```python
    tester.setStatus("why we failed", self.bucket_failed)

# to:

    tester.setStatus(self.fail, "why we failed")

# or, if no message is required (hence the change to this more simple design):
    tester.setStatus(self.fail)
```

### CODE REVIEW

File: python/TestHarness/testers/Tester.py [hunk 1]
```diff
@@ -452,44 +386,37 @@ def checkRunnableBase(self, options):
...
-            self.success_message = 'DRY RUN'
-            self.setStatus(self.success_message, self.bucket_success)
-
         # Check if we only want to run failed tests
         if options.failed_tests and options.results_storage is not None:
             result_key = options.results_storage.get(self.getTestDir(), {})
             status = result_key.get(self.getTestName(), {}).get('FAIL', '')
             if not status:
-                self.setStatus('not failed', self.bucket_silent)
+                self.setStatus(self.bucket_silent)
```
Comment (milljm) [inline python/TestHarness/testers/Tester.py]: fix this

File: python/TestHarness/schedulers/QueueManager.py [hunk 2]
```diff
@@ -25,565 +36,322 @@ def validParams():
...
+            self._finishAllJobs(job_data)
+            is_ready = False
+
+        # results file exists, but job was launched using a different scheduler
+        if (job_data.json_data.get(job_data.job_dir, {})
+            and not job_data.json_data[job_data.job_dir].get(job_data.plugin, {})):
+            self._finishAllJobs(job_data)
+            is_ready = False
+
+        return is_ready
```
Comment (milljm) [inline python/TestHarness/schedulers/QueueManager.py]: refactor this method a bit more. if elif elif, or just return False within each to same cpu cycles.

File: python/TestHarness/schedulers/Scheduler.py [hunk 3]
```diff
@@ -449,217 +271,160 @@ def satisfyLoad(self):
...
-                # Queue this new batch of runnable jobs
-                self.queueJobs(run_jobs=next_job_group)
+                # Stop the long running timer
+                job.report_timer.cancel()
 
-            # Not enough slots to run the job, currently
+                # All done
+                with j_lock:
+                    job.setStatus(job.finished)
+                    #print 'RUNNER FINISHED JOB: %s' % (job.getTestName())
```
Comment (milljm) [inline python/TestHarness/schedulers/Scheduler.py]: remove debug print

**Comment by milljm** [comment]
Mammoth is failing due to that application outputting bad unicode characters. This is now an issue, because we need to save output when performing PBS operations (output is stored in json format).

**Comment by permcody** [comment]
I've opened an issue for @YaqiWang to look at this. It's there because of some "bad practice" debugging files that are being written. Those should just be removed or use some other form of logging that's more robust. We can serialize these tests if we need this to go in sooner.

**Comment by milljm** [comment]
I have additional work I want to add to this PR

**Comment by milljm** [comment]
I found an issue with MAX FAILURES. Investigating...

**Comment by permcody** [comment]
@idaholab/moose-team - Jason has been working on Refactoring his own work for quite awhile. This PR is too big to review thoroughly and too critical to just merge. If anyone wants to pull this branch and play around with it, that would be very valuable.

**Review by permcody — APPROVED** [review]

**Pull request merged**